### PR TITLE
build: add jenkinsfile and flake file

### PR DESCRIPTION
**Pull request by markoburcul**
To be able to deploy site we need a pipeline to build it and publish it to a branch from which our caddy-git service will pull for changes. All of this will be done using nix flake to ease out the build/development process.

Referenced issue: https://github.com/status-im/status-network-docs/issues/1

# What does this PR resolve? 🚀
- build of the website content which is needed for deploying it

# Details 📝
Added nix flake file which is used in the Jenkins pipeline but can be also used by developers by just issuing `nix develop` command within the repository. It  ensures consistency across environments by specifying exact dependencies and versions of packages.

# Checklist ✅
- [ ] I have merged the main branch into my branch and resolved all conflicts
- [ ] I have documented the new code
- [x] I have smoke tested the new code locally
- [x] I have assigned the PR to myself
- [x] I have named the PR correctly

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env groovy
+library 'status-jenkins-lib@v1.8.16'
```
Comment (jakubgs) [inline Jenkinsfile:2]: I think we are already on `1.9.12`.

**Review by jakubgs — APPROVED** [review]
Would be good to also update `README` with info on where the builds are, like here:
https://github.com/waku-org/waku.org?tab=readme-ov-file#cicd

File: Jenkinsfile (hunk 1 above)
Comment (markoburcul) [reply Jenkinsfile]: Updated it to 1.9.13 since this is the last commit

**Comment by jakubgs** [comment]
Lets fix this repository to use `master` branch like all other repos in our orgs.

**Pull request merged**